.cbmc-batch/source/proof_allocators.c

```diff
@@ -105,3 +107,131 @@ void *can_fail_realloc(void *ptr, size_t newsize) {
     }
     return nondet_bool() ? NULL : realloc(ptr, newsize);
 }
+
+/************************************************************************************************************
+ * Overrides for allocator.c - for use in CBMC proofs.
+ * These assert that the allocator is can_fail_allocator, and then do the allocations without using
+ * function pointers. (Function pointers make CBMC slow).
+ */
+
+/**
+ * This assert will fail if code ever uses a different allocator than expected during a proof
+ */
+bool aws_allocator_is_valid(const struct aws_allocator *alloc) {
+    return alloc == can_fail_allocator();
+}
+
+void *aws_mem_acquire(struct aws_allocator *allocator, size_t size) {
+    AWS_FATAL_PRECONDITION(aws_allocator_is_valid(allocator));
+    /* Protect against https://wiki.sei.cmu.edu/confluence/display/c/MEM04-C.+Beware+of+zero-length+allocations */
+    AWS_FATAL_PRECONDITION(size != 0);
+
+    void *mem = can_fail_malloc(size);
+    if (!mem) {
+        aws_raise_error(AWS_ERROR_OOM);
+    }
+    return mem;
+}
+
+void *aws_mem_calloc(struct aws_allocator *allocator, size_t num, size_t size) {
+    AWS_FATAL_PRECONDITION(aws_allocator_is_valid(allocator));
+    /* Protect against https://wiki.sei.cmu.edu/confluence/display/c/MEM04-C.+Beware+of+zero-length+allocations */
+    AWS_FATAL_PRECONDITION(num != 0 && size != 0);
+
+    /* Defensive check: never use calloc with size * num that would overflow
+     * https://wiki.sei.cmu.edu/confluence/display/c/MEM07-C.+Ensure+that+the+arguments+to+calloc%28%29%2C+when+multiplied%2C+do+not+wrap
+     */
+    size_t required_bytes;
+    if (aws_mul_size_checked(num, size, &required_bytes)) {
+        return NULL;
+    }
+
+    /* If there is a defined calloc, use it */
```

Comment:
There appears to always be a defined calloc :)

Reply:
Fixed.